arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -375,30 +376,31 @@ std::shared_ptr<Index> RocksDBCollection::createIndex(VPackSlice const& info,
             << "' but found conflicting index '" << builder.slice().toJson() << "'";
 #endif
         THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_ARANGO_DUPLICATE_IDENTIFIER,
-                                       "duplicate value for `" + 
-                                           arangodb::StaticStrings::IndexId +
-                                           "` or `" + 
-                                           arangodb::StaticStrings::IndexName +
-                                           "`");
+                                       "duplicate value for `" + StaticStrings::IndexId +
+                                           "` or `" + StaticStrings::IndexName + "`");
       }
     }
   }
-
-  auto buildIdx =
-      std::make_shared<RocksDBBuilderIndex>(std::static_pointer_cast<RocksDBIndex>(idx));
-  // Step 3. add index to collection entry (for removal after a crash)
-  if (!engine->inRecovery()) {  // manually modify collection entry, other
-                                // methods need lock
-    RocksDBKey key;             // read collection info from database
-    key.constructCollection(_logicalCollection.vocbase().id(),
-                            _logicalCollection.id());
-    rocksdb::PinnableSlice ps;
-    rocksdb::Status s = engine->db()->Get(rocksdb::ReadOptions(),
-                                          RocksDBColumnFamily::definitions(),
-                                          key.string(), &ps);
-    if (!s.ok()) {
-      res.reset(rocksutils::convertStatus(s));
-    } else {
+  
+  do {
```

Comment:
As far as I can see, the `do...while(false)` construct here is only used to `break` out of it prematurely.
As the block is relatively long, wouldn't this justify moving it to a function/method of its own? That would allow `return`ing prematurely and doesn't require a `do..while(false)` workaround.

Reply:
I agree. The `do...while(false)` construct feels pretty hacky and anti-pattern here. I'd say either break it out into a function and use returns, or use a status variable and normal conditional logic.